File: DSA/Arrays/Code/3_1DArrays.c
#include <stdio.h>

// 1D Arrays in C
// Single dimensional arrays are linear arrays with one row

int main() {
    // Declaration and initialization
    int arr[5] = {10, 20, 30, 40, 50};
    
    // Accessing elements
    printf("Element at index 0: %d\n", arr[0]);
    printf("Element at index 2: %d\n", arr[2]);
    
    // Modifying elements
    arr[1] = 25;
    printf("Modified element at index 1: %d\n", arr[1]);
    
    // Iterating and printing all elements
    printf("All elements: ");
    for (int i = 0; i < 5; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");
    
    // Finding sum
    int sum = 0;
    for (int i = 0; i < 5; i++) {
        sum += arr[i];
    }
    printf("Sum of all elements: %d\n", sum);
    
    return 0;
}

File: DSA/Arrays/Code/3_1DArrays.py
# 1D Arrays in Python
# Single dimensional arrays/lists store elements linearly

# Declaration and initialization
arr = [10, 20, 30, 40, 50]

# Accessing elements
print("Element at index 0:", arr[0])
print("Element at index 2:", arr[2])

# Modifying elements
arr[1] = 25
print("Modified element at index 1:", arr[1])

# Iterating and printing all elements
print("All elements:", arr)

# Finding sum
total = sum(arr)
print("Sum of all elements:", total)

# Finding maximum and minimum
print("Maximum element:", max(arr))
print("Minimum element:", min(arr))

# Array operations
arr.append(60)  # Add element at end
print("After appending 60:", arr)

arr.insert(2, 35)  # Insert at specific index
print("After inserting 35 at index 2:", arr)
